Lib/tempfile.py

```diff
@@ -500,6 +503,15 @@ def __enter__(self):
     def __exit__(self, exc, value, tb):
         result = self.file.__exit__(exc, value, tb)
         self.close()
+        # If the file is to be deleted, but not on close, delete it now.
+        if self.delete and not self.delete_on_close:
+            try:
+                _os.unlink(self.name)
+            # It is okay to ignore FileNotFoundError. The file may have
+            # been deleted already.
+            except FileNotFoundError:
+                pass
+
```

Comment:
A case hasn't been covered. If `delete` is true and `delete_on_close` is false, but the file is never used in a `with` context, then it won't be deleted automatically when the instance is finalized. A `__del__()` method is needed for this case.

Reply:
@eryksun , I am just trying to figure out whether this is the issues introduced by my changes or this is an issue, which existed before?

Reply:
It's a new case to cover. Before there was just `delete` and `delete_on_close` was implied. On POSIX, when `delete_on_close` is true, finalizing the `_TemporaryFileCloser` instance deletes the file. On Windows, the OS deletes the file. OTOH, when `delete` is true and `delete_on_close` is false, the file has to be deleted when the `_TemporaryFileWrapper()` instance is finalized, in `__del__()`.

Reply:
OK, I think I understand now. Also the new test case needs to be created to cover this, I think. Correct?

Reply:
This is now implemented in the new [PR 97015](https://github.com/python/cpython/pull/97015)